### DISCUSSION TITLE
Feedback on COMPOSER_ALLOW_SUPERUSER

### DISCUSSION THREAD

**Discussion by Sainan** [composer/composer#9823] [General]
So, you decided to add "Continue as root/super user [yes]?" and force me to press enter on that. Great, so, I decided to skim your code and ended up using `export COMPOSER_ALLOW_SUPERUSER=1` ... but now I realized this is actually causing me to be _less secure_, because it skips the "Silently clobber any sudo credentials" portions. So, if you truly care about security, maybe these should run in any case? ...and I'll also be glad to explain why I won't stop using Linux as root anytime soon. 🙃

**Comment by Seldaek** [comment]
If you set the env var, it means you are aware of the situation and the risks, and chose to ignore it.

Alternatively if you only want to suppress the prompt but not the clobber, you can use --no-interaction / -n.

**Comment by stof** [comment]
If you are already running as root, not clobbering sudo credentials does not make you more risky: scripts don't need to have any sudo credentials to gain root priviledges, as they _already_ have it
